pkg/engine/task/task.go

```diff
@@ -0,0 +1,69 @@
+package task
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+)
+
+// Tasker is an interface that represents any runnable task for an operator. This method is treated
+// as idempotent and will be called multiple times during the life-cycle of the plan execution.
+// Method returns a boolean, signalizing that the task has finished successfully, and an error.
+// An error can wrap the ErrFatalExecution for errors that can not be retried e.g. failed template
```

Comment:
can not be retried -> should not be retried - I mean, technically you can retry 🤷‍♀ :D